### PR TITLE
If zsync eats dirt during an OTA update, allow to fallback to a full download

### LINKED ISSUE #4429: OTA update fails on Kobo Glo HD
* KOReader version:
201812000031
updating to:
2018000033

* Device: Kobo Glo HD, N437

#### Issue
I can not update koreader via OTA, the process stops with an error message
"Error updating KOReader. Would you like to delete temporary files?"

#### Steps to reproduce
1) I did an OTA update 4-5 days ago to 201812000031
2) I launch OTA update today
OTA update asks:
Do you want to update?
Installed version: 201812000031
Available version: 201812000033
3) I click UPDATE button
4) "calculating local distribution";
5) "computing zsync delta";
6) "Error updating KOReader. Would you like to delete temporary files?"
7) I click YES - no update happened; next time I "check for update" I see the same versions detected/offered;
8) I changed OTA server from
http://ota.koreader.rocks:80 to
http://vislab.bjmu.edu.cn:80/apps/koreader/ota
and got the same result
9) the third OTA server responded with something "no updates found": http://koreader.ak-team.com:80

Thank you for the proper OTA update mechanism though(when zsync started to be incorporated - since then, OTA went smoother), koreader is growing fast! thank you for your efforts

### PR DESCRIPTION
**Pull request by NiLuJe**
Fix #4429 (and by "fix", I mean "workaround' ;p).

Basically gives the user another choice on failure, one that'll download the full OTA package.
Prevents leaving users in the lurch in case zsync goes wonky again, like in #4429 ;).

### CODE REVIEW

**Comment by NiLuJe** [comment]
Last PR of 2018 \o/

File: frontend/ui/otamanager.lua [hunk 1]
```diff
@@ -175,9 +176,43 @@ function OTAManager:fetchAndProcessUpdate()
                         if self.can_pretty_print then
                             os.execute("./fbink -q -y -7 -pm ' '  ' '")
                         end
-                        UIManager:show(ConfirmBox:new{
-                            text = _("Error updating KOReader. Would you like to delete temporary files?"),
-                            ok_callback = function()
+                        UIManager:show(MultiConfirmBox:new{
+                            text = _("Failed to update KOReader.\n\nYou can:\nCancel, keeping temporary files intact.\nRetry the update process but this time, with a full download.\nAbort and cleanup all temporary files."),
```
Comment (Frenzie) [inline frontend/ui/otamanager.lua]: It can probably be slightly shorter without losing anything:
```suggestion
                            text = _("Failed to update KOReader.\n\nYou can:\nCancel, keeping temporary files.\nRetry the update process with a full download.\nAbort and cleanup all temporary files."),
```

**Comment by NiLuJe** [comment]
And before I forget once again, the reasoning behind leaving the FBInk strings untranslated:

The default font doesn't have a terribly wide coverage (Latin & Greek, basically), which is problematic.
Making it i18n friendly would imply switching to using our own fallback TTF font, and I'd probably want to do that in pure Lua via FFI because the overhead would be much lower this way for that codepath.
But we'd still need the standalone fbink binary for the startup scripts, which are already untranslated/untranslateable, and to handle TTF fonts & FFI, we'd need to switch to a full & dynamically linked FBInk build.

TL;DR: Meh.
